src/connectdlg.cpp

```diff
@@ -923,3 +933,27 @@ void CConnectDlg::DeleteAllListViewItemChilds ( QTreeWidgetItem* pItem )
         delete pCurChildItem;
     }
 }
+
+void CConnectDlg::contextMenuRequest()
+{
+    // add line to context menu
+gfContextMenuRequest( cbxServerAddr, qaDeleteAction );
+}
+
+void CConnectDlg::deleteServer()
```

Comment:
Could you pass the array reference to this method, so it can be reused, as otherwise there's quite a bit of duplicated logic.